File: library/src/main/java/android/support/design/widget/FixedCollapsingToolbarLayout.java
package android.support.design.widget;

import android.content.Context;
import android.support.v4.view.WindowInsetsCompat;
import android.util.AttributeSet;

/**
 * Created by Hakeem on 8/10/17.
 */

public class FixedCollapsingToolbarLayout extends CollapsingToolbarLayout {

    public FixedCollapsingToolbarLayout(Context context) {
        this(context, null);
    }

    public FixedCollapsingToolbarLayout(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    @Override
    protected void onMeasure(int widthMeasureSpec, int heightMeasureSpec) {
        WindowInsetsCompat oldInsets = mLastInsets;
        if (mLastInsets != null) {
            mLastInsets = mLastInsets.consumeSystemWindowInsets();
        }
        super.onMeasure(widthMeasureSpec, heightMeasureSpec);
        mLastInsets = oldInsets;
    }
}

File: library/src/main/java/android/support/design/widget/FixedCollapsingToolbarLayout.kt
package android.support.design.widget

import android.content.Context
import android.util.AttributeSet

/**
 * Created by Hakeem on 8/10/17.
 */

class FixedCollapsingToolbarLayout @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null) : CollapsingToolbarLayout(context, attrs) {

    override fun onMeasure(widthMeasureSpec: Int, heightMeasureSpec: Int) {
        val oldInsets = mLastInsets
        mLastInsets?.let {
            mLastInsets = it.consumeSystemWindowInsets()
        }
        super.onMeasure(widthMeasureSpec, heightMeasureSpec)
        mLastInsets = oldInsets
    }

}
